silk/tests/openthread/ot_test_off_mesh_route_traffic.py

```diff
@@ -0,0 +1,233 @@
+
+    @classmethod
+    @testcase.setup_class_decorator
+    def setUpClass(cls: 'TestOffMeshRouteTraffic'):
+        # Check and clean up wpantund process if any left over
+        process_cleanup.ps_cleanup()
+
+        cls.hardware_select()
+
+        for device in cls.all_nodes:
+            device.set_logger(cls.logger)
+            cls.add_test_device(device)
+            device.set_up()
+
+        cls.network_data = WpanCredentials(network_name="SILK-{0:04X}".format(random.randint(0, 0xffff)),
+                                           psk="00112233445566778899aabbccdd{0:04x}".format(random.randint(0, 0xffff)),
+                                           channel=random.randint(11, 25),
+                                           fabric_id="{0:06x}dead".format(random.randint(0, 0xffffff)))
+
+        cls.thread_sniffer_init(cls.network_data.channel)
+
+    @classmethod
+    @testcase.teardown_class_decorator
+    def tearDownClass(cls: 'TestOffMeshRouteTraffic'):
+        for device in cls.device_list:
+            device.tear_down()
+
+    @testcase.setup_decorator
+    def setUp(self):
+        pass
+
+    @testcase.teardown_decorator
+    def tearDown(self):
+        pass
+
+    @testcase.test_method_decorator
+    def test01_disable_autoupdate_interface_address_on_ncp(self):
+        for node in self.all_nodes:
+            # Disable `AutoUpdateInterfaceAddrsOnNCP` feature on wpantund
+            # for all nodes. This ensures that added IPv6 address (on linux
+            # interface) are not pushed to NCP (and therefore are not
+            # on-mesh).
+            node.setprop("Daemon:IPv6:AutoUpdateInterfaceAddrsOnNCP", "false")
+            self.assertEqual(node.getprop("Daemon:IPv6:AutoUpdateInterfaceAddrsOnNCP"), "false")
+
+    @testcase.test_method_decorator
+    def test02_pairing(self):
+        # allowlisting between leader and router
+        self.r1.allowlist_node(self.r2)
+        self.r2.allowlist_node(self.r1)
+
+        # allowlisting between leader and end device
+        self.r1.allowlist_node(self.fed1)
+        self.fed1.allowlist_node(self.r1)
+
+        # allowlisting between router and sleepy-end-device
+        self.r2.allowlist_node(self.sed2)
+        self.sed2.allowlist_node(self.r2)
+
+        self.r1.form(self.network_data, "router")
+        self.r1.permit_join(60)
+        self.wait_for_completion(self.device_list)
+
+        self.logger.info(self.r1.ip6_lla)
+        self.logger.info(self.r1.ip6_thread_ula)
+
+        self.network_data.xpanid = self.r1.xpanid
+        self.network_data.panid = self.r1.panid
+
+        self.r2.join(self.network_data, "router")
+        self.wait_for_completion(self.device_list)
+
+        self.fed1.join(self.network_data, "end-node")
+        self.wait_for_completion(self.device_list)
+
+        self.sed2.join(self.network_data, "sleepy-end-device")
+        self.sed2.set_sleep_poll_interval(POLL_INTERVAL)
+        self.wait_for_completion(self.device_list)
+
```

Comment:
Pls add code here to check if r2 becomes a "router"

Reply:
Ack